<!-- Runs obstacle tracker with a virtual obstacle publisher. -->
<launch>
  <node pkg="tf" type="static_transform_publisher" name="world_to_scanner" output="screen" args="0 0 0 0 0 0 world scanner 10"/>
  
  <node pkg="obstacle_detector" type="obstacle_publisher" name="obstacle_publisher">
    <param name="active"           value="true"/>
    <param name="reset"            value="false"/>
    <param name="fusion_example"   value="false"/>
    <param name="fission_example"  value="true"/>

    <param name="loop_rate"        value="10.0"/>

    <param name="frame_id"         value="world"/>

    <remap from="obstacles" to="raw_obstacles"/>
  </node>
 
  <node pkg="obstacle_detector" type="obstacle_tracker" name="obstacle_tracker" output="screen">
    <param name="active"                  value="true"/>

    <param name="loop_rate"               value="100.0"/>
    <param name="tracking_duration"       value="2.0"/>
    <param name="min_correspondence_cost" value="0.3"/>
    <param name="std_correspondence_dev"  value="0.15"/>
    <param name="process_variance"        value="0.001"/>  
    <param name="process_rate_variance"   value="0.01"/>  
    <param name="measurement_variance"    value="1.0"/>  

    <remap from="tracked_obstacles" to="obstacles"/>
  </node>

  <node pkg="obstacle_detector" type="obstacle_visualizer" name="obstacle_visualizer" output="screen">
    <param name="tracked_circles_color"   value="3"/>
    <param name="untracked_circles_color" value="2"/>
    <param name="segments_color"          value="7"/>

    <param name="active"                  value="true"/>
    <param name="show_labels"             value="true"/>

    <param name="alpha"                   value="0.7"/>
    <param name="z_layer"                 value="0.0"/>
  </node>

  <node pkg="obstacle_detector" type="obstacle_recorder" name="obstacle_recorder" output="screen">
    <param name="active"                  value="true"/>
    <param name="recording"               value="false"/>

    <param name="filename_prefix"         value="tracked_"/>
  </node>

  <node pkg="rviz" type="rviz" name="rviz" output="screen" args="-d $(find obstacle_detector)/resources/obstacle_detector.rviz"/>
</launch>

